pkg/provider/aws/config.go

```diff
@@ -34,8 +34,9 @@ const (
 
 var (
 	eksDefaults = config.KubernetesTypeParams{
-		NodeType: "fargate",
-		Replicas: 2,
+		NodeType:     "fargate",
+		InstanceType: "t3.medium",
```

Comment:
this combination doesnt quite make sense since you cant choose instance type on fargate. Not sure if it would actually make our logic have issues though, but its a bit confusing

Reply:
Yeah, but there's no "per-nodetype default" configuration, so this is as good as we can do for now.

Reply:
So i think we actually dont want this default expose, but only internally in our plugin. the reasoning being, if someone sets an instanceType on one unit but not the other and that instance type isnt t3.medium, this would force us to create 2 node groups which would be wasteful. Instead we should know that because the unit2 doesnt have an instance type or disk space specified, it can launch on any node

Reply:
if we want to set this it would need to be at the cluster level now that i think about it

Reply:
> if someone sets an instanceType on one unit but not the other and that instance type isnt t3.medium, this would force us to create 2 node groups which would be wasteful.

The same could be said about many other parameters. If the intention is to change the instance type for all units, we provide a mechanism for that via the `defaults`. I think this should behave as all the other parameters and not special cased.

Reply:
its not special cased, its just a different scenario. If i was a user and i didnt specify anything, i wouldnt expect a node group to be created for a specific execution unit which didnt have needs for instance type or disk size because there are other node groups which satisfy its constraints.

I also just dont think we should be regressing in terms of optimizing architectures for cost

Reply:
a node doesnt belong to a kubernetes exec unit, like it does to an rds instance, etc